bundles/org.openhab.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -188,9 +188,13 @@
     void handleRemoval();
 
     /**
-     * This method provides a list of classes which should be registered as services by the framework
+     * This method provides a list of classes which should be registered as services by the framework.
+     * <p>
+     * Use this method for example to return your thing specific discoveries (extending {@link AbstractDiscoveryService}).
```

Comment:
Hm, using the `AbstractDiscoveryService` this way is IMHO more complicated as you need to provide other services. Didn't we?

Reply:
But `getServices()` was to my knowledge purely created to support this use-case. If the implementation is un-usable (I mean it really is), then we should replace it instead.

I would say we use `@Component(scope=ServiceScope.PROTOTYPE)` on those services and let the framework create them and not creating our own object instance first via newInstance.

Reply:
I was not aware that we ever tried to create a new component logic.
For me it seems it has been created for simpler services like the ones it has been contributed: https://github.com/eclipse/smarthome/pull/6398

Reply:
But don't you occasionally need component dependencies in your thingActions as well?
I'm using this for discovery services tied to things because I was told to by other core maintainers and I liked that simpler approach. But if that is not the intention of the API, the documentation should be adopted to say so.